bootable-jar/WFLY-14934_Server_Mode.adoc

```diff
@@ -0,0 +1,128 @@
+= Server Mode for Bootable JAR Maven Plugin
+:author:           Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Introduce a mode in the Bootable JAR Maven Plugin to build a server installation with the application deployment. 
+This mode is an alternative to the existing mode that bundles the server installation and the application deployment inside 
+an executable Bootable JAR.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14934[WFLY-14934]
+
+=== Related Issues
+
+* New WildFly S2I: https://issues.redhat.com/browse/WFLY-14936[WFLY-14936]
+
+* JKube evolved support: https://issues.redhat.com/browse/WFLY-14936[WFLY-14938]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin/[Bootable JAR Maven plugin]
+
+* https://github.com/eclipse/jkube[JKube]
+
+* https://github.com/wildfly/wildfly-s2i[WildFly s2i]
+
+* https://github.com/wildfly/quickstart[WildFly quickstarts]
+
+* ODO dev files.
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [ ] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
```

Comment:
Since this feature seems to be OpenShift/S2I focused we need to ensure it is microservices friendly without antipatterns like multiple deployments.